addons/web/static/tests/services/data_manager_tests.js

```diff
@@ -0,0 +1,130 @@
+odoo.define('web.data_manager_tests', function (require) {
+    "use strict";
+
+    const DataManager = require('web.DataManager');
+    const rpc = require('web.rpc');
+    const config = require('web.config');
+    const testUtils = require('web.test_utils');
+
+    /**
+     * Create a simple data manager with mocked functions:
+     * - mockRPC -> rpc.query
+     * - isDebug -> config.isDebug
+     * @param {Object} params
+     * @param {Function} params.isDebug
+     * @param {Function} params.mockRPC
+     * @returns {DataManager}
+     */
+    function createDataManager({ isDebug, mockRPC }) {
+        testUtils.mock.patch(rpc, { query: mockRPC });
+        testUtils.mock.patch(config, { isDebug });
+
+        const dataManager = new DataManager();
+        return dataManager;
+    }
+
+    /**
+     * Helper function returning an object similar to the return value of a server
+     * 'load_views'.
+     * @param {Object} params
+     * @param {Object} [params.args]
+     * @param {Object} params.kwargs
+     * @param {Object} [params.model]
+     * @returns {Object}
+     */
+    function mockLoadViews({ args, kwargs, model }) {
+        const result = {
+            fields: {},
+            fields_views: {},
+        };
+        kwargs.views.forEach(([viewId, viewType]) => {
+            result.fields_views[viewType] = {
+                model,
+                arch: `<${viewType}/>`,
+                view_id: viewId,
+                type: viewType,
+                fields: {},
+            };
+        });
+        if (kwargs.options.load_filters) {
+            result.filters = [
+                { id: 3, name: "aaa", context: "{}", domain: "[]" },
+                { id: 6, name: "bbb", context: "{}", domain: "[]" },
+            ];
+        }
+        return result;
+    }
+
+    QUnit.module("Services", {
+        beforeEach() {
+            this.loadViewsParams = {
+                model: "ir.oui",
+                context: {},
+                views_descr: [
+                    [10, 'kanban'],
+                    [20, 'list'],
+                    [30, 'search'],
+                ],
+            };
+        },
+        afterEach() {
+            testUtils.mock.unpatch(rpc);
+            testUtils.mock.unpatch(config);
```

Comment:
i'd patch in beforeEach then, by consistency

Reply:
The methods used in the patches are provided by each individual test

Reply:
ok